superset-frontend/src/explore/components/controls/VizTypeControl.jsx

```diff
@@ -166,15 +166,21 @@ const VizTypeControl = props => {
   const filterString = filter.toLowerCase();
 
   const filteredTypes = DEFAULT_ORDER.filter(type => registry.has(type))
-    .filter(type => !registry.get(type).isNativeFilter)
+    .filter(type => {
+      const behaviors = registry.get(type)?.behaviors || [];
+      return behaviors.includes(Behavior.CROSS_FILTER) || !behaviors.length;
+    })
     .map(type => ({
       key: type,
       value: registry.get(type),
     }))
     .concat(
       registry
         .entries()
-        .filter(entry => !entry.value.isNativeFilter)
+        .filter(entry => {
```

Comment:
may be move it to some utils like
```
getVisibleCharts = ({ behaviours } = {}) => {
      return behaviors?.includes(Behavior.CROSS_FILTER) || !behaviors?.length;
    }
}
and use it like:

```
.filter(type =>getVisibleCharts(registry.get(type)))
and
.filter(entry =>getVisibleCharts(entry.value))
```